src/plugin/indexer-solr/src/java/org/apache/nutch/indexwriter/solr/SolrIndexWriter.java

```
@@ -75,19 +70,61 @@
   private int totalUpdates = 0;
   private boolean delete = false;
 
+  @Override
   public void open(JobConf job, String name) throws IOException {
-    solrClients = SolrUtils.getSolrClients(job);
-    init(solrClients, job);
+    //Implementation not required
   }
 
-  // package protected for tests
-  void init(List<SolrClient> solrClients, JobConf job) throws IOException {
-    batchSize = job.getInt(SolrConstants.COMMIT_SIZE, 1000);
-    solrMapping = SolrMappingReader.getInstance(job);
-    delete = job.getBoolean(IndexerMapReduce.INDEXER_DELETE, false);
+  /**
+   * Initializes the internal variables from a given index writer configuration.
+   *
+   * @param parameters Params from the index writer configuration.
+   * @throws IOException Some exception thrown by writer.
+   */
+  @Override
+  public void open(Map<String, String> parameters) throws IOException {
+    String type = parameters.getOrDefault("type", "http");
+
+    String[] urls = StringUtils.getStrings(parameters.get("url"));
+
+    if (urls == null) {
+      String message = "Missing SOLR URL.\n" + describe();
+      LOG.error(message);
+      throw new RuntimeException(message);
+    }
+
+    this.solrClients = new ArrayList<>();
+
+    switch (type) {
+      case "http":
+        for (String url : urls) {
+          solrClients.add(SolrUtils.getHttpSolrClient(url));
+        }
+        break;
+      case "cloud":
+        for (String url : urls) {
+          CloudSolrClient sc = SolrUtils.getCloudSolrClient(url);
+          sc.setDefaultCollection(parameters.get(SolrConstants.COLLECTION));
+          solrClients.add(sc);
+        }
+        break;
+      case "concurrent":
```

Can you throw unsupported Exception at this stage? and also a default case?